contracts/end.cairo

```diff
@@ -834,8 +844,20 @@ func thaw{
     let (cure_debt) = CureLike.tell(cure);
     let (new_debt) = sub(vat_debt, cure_debt);
     _debt.write(new_debt);
+
+    let (self) = get_contract_address();
+    let (claim) = _claim.read();
+
+    // claim.mint(address(this), debt);
+    ClaimLike.mint(claim, self, new_debt);
+
+    // claim.approve(vow, type(uint256).max);
+    let (max) = _felt_to_uint(2 ** 255);
```

Comment:
`2**255` is not what you think. We have `uint256.max` somewhere. `_felt_to_uint` feels like antipattern.

Reply:
`uint256.max` = `Uint256(2 ** 128 - 1, 2 ** 128 - 1)`